apps/emqx_retainer/src/emqx_retainer_mnesia_cli.erl

```diff
@@ -50,26 +50,17 @@ retainer(["reindex", "status"]) ->
 retainer(["reindex", "start"]) ->
     retainer(["reindex", "start", "false"]);
 retainer(["reindex", "start", ForceParam]) ->
-    Force =
-        case ForceParam of
-            "true" -> true;
-            _ -> false
-        end,
-    ?PRINT_MSG("Starting reindexing~n"),
-    emqx_retainer_mnesia:reindex(
-        Force,
-        fun(Done) ->
-            ?SLOG(
-                info,
-                #{
-                    msg => "retainer_message_record_reindexing_progress",
-                    done => Done
-                }
-            ),
-            ?PRINT("Reindexed ~p messages~n", [Done])
-        end
-    ),
-    ?PRINT_MSG("Reindexing finished~n");
+    case mria_rlog:role() of
```

Comment:
I think this check is not needed.  `mria_lib:call_backend_rw_dirty` already performs this check and routes the operation to a core node.

Reply:
We iterate over retained messages through `ets`, so we should be sure it is up to date.